### ISSUE TITLE
Certificate authentication regression

### ISSUE THREAD

**Issue by intgr** [hvac/hvac#721]
Updating to hvac version 0.10.14 broke certificate authentication in my use case.

The regression seems to be introduced in 0.10.12, In PR #691. It's clear that this did not receive sufficient review. There are many parts of the code that clearly don't do what the developer intended -- added a few review comments there. And I also submitted a minor pull request #720.

The fact that this one change has already received 2 fix releases and continues to cause problems, IMO indicates insufficient quality control.

I reduced our code down to this reproducible snippet:

```python
import hvac

cert_file = '/.../..._client.crt.pem'
pkey_file = '/.../..._client.key.pem'
url = "https://XXXXX/"

client = hvac.Client(
    url=url,
    verify=True,
    cert=(cert_file, pkey_file),
)
client.auth_tls()
print(f'token length:', len(client.token))
```

Result:
```
% pip install hvac==0.10.11
Collecting hvac==0.10.11
  Using cached hvac-0.10.11-py2.py3-none-any.whl (139 kB)
...
% python ./_local/hvacbug.py
token length: 26
```
```
% pip install hvac==0.10.14
Collecting hvac==0.10.14
  Using cached hvac-0.10.14-py2.py3-none-any.whl (142 kB)
...
% python ./_local/hvacbug.py
.../_local/hvacbug.py:12: DeprecationWarning: Call to deprecated function 'auth_tls'. This method will be removed in version '0.13.0' Please use the 'login' method on the 'hvac.api.auth_methods.cert' class moving forward.
  client.auth_tls()
Traceback (most recent call last):
  File "/.../_local/hvacbug.py", line 12, in <module>
    client.auth_tls()
...
  File "/.../lib/python3.9/site-packages/hvac/utils.py", line 37, in raise_for_error
    raise exceptions.InvalidRequest(message, errors=errors, method=method, url=url)
hvac.exceptions.InvalidRequest: client certificate must be supplied, on post https://XXXXX/v1/auth/cert/login
```

**Comment by Tylerlhess** [comment]
The Client class doesn't save the certificate values passed during construction as part of the class. If you do the auth_tls() as part of the constructor it is still in scope and will work. This is the same behavior that existed before my rework but I would have had to rework the adapter class to fix it correctly and so figured that keeping the same functionality would work.

**Comment by intgr** [comment]
> If you do the auth_tls() as part of the constructor it is still in scope and will work.

I don't understand the answer. How would I "do the auth_tls() as part of the constructor"? Are you saying that something is wrong with my example code?

The latest hvac documentation still contains similar example code https://hvac.readthedocs.io/en/v0.11.0/usage/auth_methods/index.html#authenticate-to-different-auth-backends

```python
# TLS
client = Client(cert=('path/to/cert.pem', 'path/to/key.pem'))
client.auth_tls()
```

Which as far as I can tell is equievalent to my example.

> This is the same behavior that existed before my rework

Erm? I just showed that the behavior differs between versions 0.10.11 and 0.10.14. Do you think I am mistaken?

**Comment by Tylerlhess** [comment]
Yeah I was wrong. I was never able to get the cert to stick to the underlying requests object. I always had to call it like this.

`#TLS
client = Client(cert=('path/to/cert.pem', 'path/to/key.pem')).auth_tls()`

I was trying to set the Client certificate after it was created. Which looking back, was exactly what I was doing wrong. 

I originally was incorrectly trying to pass the cert as part of the auth_tls() and not as part of the Client constructor. This didn't work and was most of the reason I did all the work of building out this section of the python client. And as you pointed out there was a lack of oversight (I was honestly surprised there was so little push back on the submission). Which looking back at the entirety of the reason I did it was because I wanted to auth with the cert after creating the client instead of just recreating the client with the cert in the first place.
